linux-system-roles/rhc
# Implement rhc_release

Add a parameter for specifying the release to set for the system: pass it as parameter for "redhat_subscription", which will set the release, however only when registering (i.e. with a status change). When already registered, manually invoke the "rhsm_release" module to set/unset release, making sure to do it only when there is no status change (i.e. system registered, and requesting it to be registered).

The only gotcha here is that we need to pass "rhc_auth" also when invoking the "rhc" role only to disable all the repositories: this is because we unconditionally invoke the "redhat_subscription" module in the role, and that module currently require either the "username" or the "activationkey" parameter in case that "state=present" (which is the default). This will need future changes in the role to workaround this limitation, and possibly making the "redhat_subscription" module slightly smarter.

Make sure to test this:
- extend the test setup with a release variable
- add an helper script for querying in a single way the currently set release, that works like a function
- add a new test to verify that:
  - setting the release works when registering
  - setting/unsetting the release works on an already registered system
  - invalid releases are properly rejected (by subscription-manager)
- skip the test in case it is not possible to set a release
